### DISCUSSION TITLE
Timerate problem - scripting

### DISCUSSION THREAD

**Discussion by fzs1933** [Stellarium/stellarium#1601] [Q&A] [answered]
Hi,

I would like to stop the sky rotate by script at a certain time (for example: start at 2021-03-21T12:00:00 and stop at 2022-03-22T00:00:00).
It is 12 hours. I want to rotate the sky to the stop time in 12 seconds, the timerate will be 3600 (1 second is an hour).
But the result is not the excepted. In fact it is not consistent (see the picture below!).
I tried to stop it with the core.waitFor command: the result is the same - not consistent...
I would like to make a longer video about nightsky and it would make the job easier if I could stop the time exactly.
What might be the problem? Time equation? Or anything else?
Can help me anyone?
Thanks!
Zsolt

the script:
core.setDate("2021-03-21T11:00:00");
core.setTimeRate(0);
core.wait(1);
core.setTimeRate(3600);
core.wait(12);
core.setTimeRate(0);

![start_to_ends](https://user-images.githubusercontent.com/81897104/114262091-63bc3400-99de-11eb-89bf-cf5b6fd2618f.jpg)

**Comment by gzotti** [comment, marked as answer]
These commands have not been developed with movie timing in mind, but probably with Planetarium operators. You could decrease time rate shortly before the end time: WaitFor(23:50) and then Waitfor(midnight) with a slower timerate. Or WaitFor(0:10) and cut away excess frames in your cutting program.

**Reply by fzs1933** [reply]
Thanks your quick answer!

I can understand your proposal. And I will probably do it that way. 
There is one more question: 
timeRate(1) means that one second elapses in one second. If the timeRate(60), it takes sixty seconds in one second. So, I thought that the timeRate the number gives the exact seconds, because the second as a unit of time is independent of the length of the astronomical day.
But that’s just technical interest on my part anymore, because I’m wondering what’s causing this inconsistency. 
Thanks again!
Zsolt

**Reply by gzotti** [reply]
Look into Hard vs. Soft real-time systems: https://en.wikipedia.org/wiki/Real-time_computing. A deep analysis of Stellarium's run-time behaviour is unfortunately homework for somebody who wishes to fulfil hard real-time constraints.

**Reply by fzs1933** [reply]
Thanks.
I am affraid of this homework is beyond my knowledge... :)
I will do your proposal.
Have a nice weekend!
